DependencyInjection/MonologExtension.php

```diff
@@ -227,10 +228,28 @@ private function buildHandler(ContainerBuilder $container, $name, array $handler
                     ]);
                     $transport->setPublic(false);
 
+                    if (isset($handler['publisher']['encoder'])) {
+                        if ('compressed_json' === $handler['publisher']['encoder']) {
+                            $encoderClass = 'Gelf\Encoder\CompressedJsonEncoder';
+                        } elseif ('json' === $handler['publisher']['encoder']) {
+                            $encoderClass = 'Gelf\Encoder\JsonEncoder';
+                        } else {
+                            throw new InvalidConfigurationException('The gelf message encoder must be either "compressed_json" or "json".');
```

Comment:
An `InvalidConfigurationException` is usually thrown by the configuration, not the extension:

```suggestion
                            throw new \RuntimeException('The gelf message encoder must be either "compressed_json" or "json".');
```